File: Server/Utils/ThriftUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain;
namespace Thrift
{
    public class ThriftUtils
    {
        public static Domain.Cursa toModelCursa(Cursa cursa)
        {
            DateTime date;
            DateTime.TryParse(cursa.Date, out date);
            return new Domain.Cursa(cursa.Id, cursa.Destinatie, date, cursa.LocuriDisponibile);
        }
        public static Cursa fromNativeCursa(Domain.Cursa cursa)
        {
            Cursa c = new Cursa();
            c.Id = cursa.id;
            c.LocuriDisponibile = cursa.locuriDisponibile;
            c.Destinatie = cursa.destinatie;
            c.Date = cursa.date.ToString("dd-MM-yyyy HH:mm");
            return c;
        }
        public static List<Rezervare> fromNativeRezervari(List<Domain.Rezervare> rezervari)
        {
            var all = new List<Rezervare>();
            rezervari.ForEach(x =>
            {
                var r = new Rezervare();
                r.Id = x.id;
                r.Client = x.client;
                r.IdCursa = x.idCursa;
                r.NrLoc = x.nrLoc;
                all.Add(r);
            });
            return all;
        }
    }
}

File: Client/Thrift Generated Model/ThriftUtils.java
package app.clientFX.Thrift;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.util.List;

public class ThriftUtils {
    public static List<app.model.Cursa> toModelCurse(List<app.clientFX.Thrift.Cursa> curse){
        List<app.model.Cursa> all = new ArrayList<>();
        curse.stream().forEach(x->{

            LocalDateTime date = LocalDateTime.parse(x.date,DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"));
            all.add(new app.model.Cursa(x.id,x.destinatie, date,x.locuriDisponibile));
        });
        return all;
    }
    public static List<app.model.Rezervare> toModelRezervari(List<app.clientFX.Thrift.Rezervare> rezervari){
        List<app.model.Rezervare> all = new ArrayList<>();
        rezervari.stream().forEach(x->{
            all.add(new app.model.Rezervare(x.id,x.idCursa,x.nrLoc,x.client));
        });
        return all;
    }
    public static List<app.clientFX.Thrift.Rezervare> fromNative(List<app.model.Rezervare> rezervari){
        List<Rezervare> all = new ArrayList<>();
        rezervari.stream().forEach(x->{
            Rezervare r = new Rezervare();
            r.id = x.getId();
            r.idCursa = x.getIdCursa();
            r.client = x.getClient();
            r.nrLoc = x.getNrLoc();
            all.add(r);
        });
        return all;
    }
    public static app.model.Cursa toModelCursa(Cursa cursa){
        return new app.model.Cursa(cursa.id,cursa.destinatie,LocalDateTime.now(),cursa.locuriDisponibile);
    }
}
